clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.groovy

```diff
@@ -236,19 +243,27 @@ class GoogleNamedAccountCredentials implements AccountCredentials<GoogleCredenti
                                         instanceTypeDisks,
                                         consulConfig,
                                         compute,
-                                        userDataFile)
+                                        userDataFile,
+                                        regionsToIndex)
 
     }
   }
 
   public List<Map> getRegions() {
     List<Map> regionList = []
-
-    regionToZonesMap.each { String region, List<String> zones ->
-      Map regionMap = new HashMap()
-      regionMap.put("name", region)
-      regionMap.put("zones", zones)
-      regionList.add(regionMap)
+    if (regionsToIndex) {
+      regionsToIndex.each { String region ->
+        Map regionMap = new HashMap()
+        regionMap.put("name", region)
+        regionList.add(regionMap)
+      }
+    } else {
+      regionToZonesMap.each { String region, List<String> zones ->
+        Map regionMap = new HashMap()
+        regionMap.put("name", region)
+        regionMap.put("zones", zones)
```

Comment:
The `regionMap` in the `if` block is missing this key (I think it's only used by the UI, which is probably why you didn't see any issues).